stdlib/2and3/imghdr.pyi

```diff
@@ -1,16 +1,21 @@
-from typing import overload, Union, Text, BinaryIO, Optional, Any, List, Callable
-import sys
 import os
+import sys
+from typing import Any, BinaryIO, Callable, List, Optional, Protocol, Text, Union, overload
+
+class _ReadableBinary(Protocol):
+    def tell(self) -> int: ...
+    def read(self, size: int) -> bytes: ...
+    def seek(self, offset: int) -> Any: ...
 
 
 if sys.version_info >= (3, 6):
-    _File = Union[Text, os.PathLike[Text], BinaryIO]
+    _File = Union[Text, os.PathLike[Text], _ReadableBinary]
 else:
-    _File = Union[Text, BinaryIO]
+    _File = Union[Text, _ReadableBinary]
 
 
 @overload
 def what(file: _File) -> Optional[str]: ...
 @overload
 def what(file: Any, h: bytes) -> Optional[str]: ...
-tests: List[Callable[[bytes, BinaryIO], Optional[str]]]
+tests: List[Callable[[bytes, Optional[BinaryIO]], Optional[str]]]
```

Comment:
So when you run `open(file, 'rb')` the type you get back is `BinaryIO` which is why I made it so here. I think it should be ok to keep it as such but keen to hear if there's something better to use here. FWIW, `f` isn't actually used in any of the test functions unless I'm mistaken.